es5-persistence/src/main/java/com/netflix/conductor/elasticsearch/es5/EmbeddedElasticSearchV5.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.elasticsearch.es5;
+
+import com.netflix.conductor.elasticsearch.EmbeddedElasticSearch;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.node.InternalSettingsPreparer;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.transport.Netty4Plugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collection;
+
+import static java.util.Collections.singletonList;
+
+
+public class EmbeddedElasticSearchV5 implements EmbeddedElasticSearch {
+
+    private static final Logger logger = LoggerFactory.getLogger(EmbeddedElasticSearchV5.class);
+
+    private Node instance;
+    private Client client;
+    private File dataDir;
+
+    private class PluginConfigurableNode extends Node {
+        public PluginConfigurableNode(Settings preparedSettings, Collection<Class<? extends Plugin>> classpathPlugins) {
+            super(InternalSettingsPreparer.prepareEnvironment(preparedSettings, null), classpathPlugins);
+        }
+    }
+
+    @Override
+    public void start() throws Exception {
+        start(DEFAULT_CLUSTER_NAME, DEFAULT_HOST, DEFAULT_PORT, true);
+    }
+
+    public synchronized void start(String clusterName, String host, int port, boolean enableTransportClient) throws Exception {
+
+        if (instance != null && !instance.isClosed()) {
+            logger.info("Elastic Search is already running on port {}", getPort());
```

Comment:
I'm not sure this log line accomplishes much. It tries to be helpful, but seems to be masking a programming error -- calling `start` twice. Maybe we should `throw` instead, although I don't love the idea.

Reply:
I am on the fence about this as well, but I am leaning toward throwing.